### PR TITLE
Major Cleanup

### PR DESCRIPTION
**Pull request by ShoGinn**
I have tried to do the bare minimum of implementing the IDE portion.

But I did add the .gitignore and ruff editor options.

This will take a while to review, but it's worth it.

I found several possible typing situations that could cause problems, as well as getting rid of all the "Literals" and using enums.

Hopefully, it meets your coding objectives

## Summary by CodeRabbit

- **New Features**
  - Introduced configuration for the `ruff` tool in `pyproject.toml` to enhance code formatting.
  - Added new error handling classes and attributes in `twikit` for better exception management.
  - Updated `twikit` classes with new attributes and improved method signatures for enhanced functionality and clarity.
  - Introduced new enum classes for tweet-related types in `twikit/tweet.py`.

- **Bug Fixes**
  - Corrected attribute types in `twikit/trend.py` and `twikit/tweet.py` to allow `None` values.
  - Fixed typos in exception descriptions across various modules.

- **Refactor**
  - Reorganized imports and updated type annotations in `twikit` for improved readability and maintainability.
  - Enhanced cookie handling in `HTTPClient` class for better web session management.

- **Documentation**
  - Updated `.gitignore` to include rules for various development environments and files, ensuring cleaner repositories.

### CODE REVIEW

**Comment by ShoGinn** [comment]
I did not touch the async portion of the script...

File: twikit/utils.py [hunk 1]
```diff
@@ -526,16 +512,7 @@ def b64_to_str(b64: str) -> str:
...
-    'media',
-    'retweets',
-    'native_video',
-    'periscope',
-    'vine',
-    'images',
-    'twimg',
-    'links'
-]
+FILTERS = Literal['media', 'retweets', 'native_video', 'periscope', 'vine', 'images', 'twimg', 'links']
```
Comment (d60) [inline twikit/utils.py:515]: Please make the line length 100 characters or less.

File: twikit/http.py [hunk 2]
```diff
@@ -54,9 +49,8 @@ def post(self, url, **kwargs) -> httpx.Response:
...
-        cookies = {}
+        cookies = httpx.Cookies()
         for cookie in self.client.cookies.jar:
-            if 'ct0' in cookies and cookie.name == 'ct0':
-                continue
-            cookies[cookie.name] = cookie.value
-        self.client.cookies = list(cookies.items())
+            if cookie.value is not None and (cookie.name != 'ct0' or 'ct0' not in cookies):
+                cookies.set(cookie.name, cookie.value)
+        self.client.cookies = cookies
```
[the remark this answers is not in the document]
Comment (ShoGinn) [reply twikit/http.py:55]: The intent is to only remove duplicate ct0 this refactor only includes ct0

[the remark this answers is not in the document]
Comment (ShoGinn) [reply twikit/http.py:55]: This updated code is less pythonic than my code and requires extra booleans.

**Comment by ShoGinn** [comment]
@coderabbitai review

**Comment by ShoGinn** [comment]
@d60 Do we plan on merging these? If not, I'll close this and stop merging branches whenever you update.

**Pull request closed without merging**